### ISSUE TITLE
fix: `plainToClass` omitting decorated properties in Next.js

### ISSUE THREAD

**Issue by aboqasem** [typestack/class-transformer#1087]
### Description

`plainToClass` omitting properties in Next.js, same code with Nest.js works properly.

**Minimal code-snippet showcasing the problem**

Bug in Next.js: https://codesandbox.io/s/class-transformer-plaintoclass-omitting-bugged-tm6pz?file=/pages/index.tsx
Working in Nest.js: https://codesandbox.io/s/class-transformer-plaintoclass-omitting-ok-t91tr?file=/src/app.controller.ts

 

```ts
class Some {
  @IsString()
  email!: string;
}

const some = plainToClass(Some, {
  id: 1,
  email: '',
});

console.log(some, 'email' in some);
// nextjs: { id: 1, email: undefined } true
// nestjs: { id: 1, email: '' } true
```

### Expected behavior

Not to omit decorated properties like in Nest.js.

### Actual behavior

Omitting decorated properties.

**Comment by aboqasem** [comment]
**UPDATE:**

Outputs of these are different:

```ts
console.log(require('class-transformer/esm5/storage').defaultMetadataStorage); // nextjs in browser
console.log(require('class-transformer/cjs/storage').defaultMetadataStorage); // nestjs & nextjs api
```

Still do not know what is wrong. If anybody has a guess what could be the issue please tell me and I will try to contribute.

**Comment by aboqasem** [comment]
Using `git bisect` I found that this diff is what broke the app. It is `nx migrate latest` which upgrades `next` from `11.1.2` to `12.0.4`:

```diff
diff --git a/apps/web/.babelrc b/apps/web/.babelrc
deleted file mode 100644
index c7d82af..0000000
--- a/apps/web/.babelrc
+++ /dev/null
@@ -1,4 +0,0 @@
-{
-  "presets": ["@nrwl/next/babel"],
-  "plugins": []
-}
diff --git a/package.json b/package.json
index 21fe22c..61bcd02 100644
--- a/package.json
+++ b/package.json
@@ -51,7 +51,7 @@
     "@nestjs/passport": "^8.0.1",
     "@nestjs/platform-express": "8.2.0",
     "@nestjs/typeorm": "^8.0.2",
-    "@nrwl/next": "13.0.1",
+    "@nrwl/next": "13.1.4",
     "@sendgrid/mail": "7.6.0",
     "@tailwindcss/forms": "^0.3.4",
     "@types/lodash.get": "^4.4.6",
@@ -66,7 +66,7 @@
     "helmet": "^4.6.0",
     "iterare": "^1.2.1",
     "lodash.get": "^4.4.2",
-    "next": "11.1.2",
+    "next": "12.0.4",
     "nprogress": "^0.2.0",
     "passport": "^0.4.1",
     "passport-local": "^1.0.0",
@@ -91,17 +91,17 @@
     "@commitlint/config-conventional": "14.1.0",
     "@nestjs/schematics": "8.0.4",
     "@nestjs/testing": "8.2.0",
-    "@nrwl/cli": "13.0.1",
-    "@nrwl/cypress": "13.0.1",
-    "@nrwl/eslint-plugin-nx": "13.0.1",
-    "@nrwl/jest": "13.0.1",
-    "@nrwl/linter": "13.0.1",
-    "@nrwl/nest": "13.0.1",
-    "@nrwl/node": "13.0.1",
-    "@nrwl/react": "13.0.1",
-    "@nrwl/tao": "13.0.1",
-    "@nrwl/web": "13.0.1",
-    "@nrwl/workspace": "13.0.1",
+    "@nrwl/cli": "13.1.4",
+    "@nrwl/cypress": "13.1.4",
+    "@nrwl/eslint-plugin-nx": "13.1.4",
+    "@nrwl/jest": "13.1.4",
+    "@nrwl/linter": "13.1.4",
+    "@nrwl/nest": "13.1.4",
+    "@nrwl/node": "13.1.4",
+    "@nrwl/react": "13.1.4",
+    "@nrwl/tao": "13.1.4",
+    "@nrwl/web": "13.1.4",
+    "@nrwl/workspace": "13.1.4",
     "@testing-library/react": "12.1.2",
     "@testing-library/react-hooks": "7.0.2",
     "@types/bcrypt": "^5.0.0",
@@ -124,7 +124,7 @@
     "babel-jest": "27.3.1",
     "cypress": "9.0.0",
     "eslint": "7.32.0",
-    "eslint-config-next": "11.1.2",
+    "eslint-config-next": "12.0.4",
     "eslint-config-prettier": "^8.3.0",
     "eslint-plugin-cypress": "^2.12.1",
     "eslint-plugin-import": "2.25.3",
```

**Comment by aboqasem** [comment]
https://github.com/typestack/class-transformer/blob/b2430dc9a56344ff44721ea7f4f1ca60dc31ec3b/src/TransformOperationExecutor.ts#L296-L307

While debugging the issue; `descriptor` here is:

```jsonc
// next@12.0.4 - next@12.0.11-canary.15:
{
  "writable": true,
  "enumerable": true,
  "configurable": true
}
// next@11.1.2 & nestjs:
undefined
```

Thus, the block is being run and the `email` is being skipped in `next@12.0.4` through `next@12.0.11-canary.15`.

**Comment by aboqasem** [comment]
Looks like this should fix this: swc-project/swc/pull/3459. But Next.js hasn't upgraded to swc v1.2.137 yet.
